### PR TITLE
[REJECTED] Mention if symbol isDeprecated

### LINKED ISSUE #12730: Error messages should include deprecation status of symbol
## Reproduction steps

Scala version: every version

```scala
scala> 42.formatted("%d")
          ^
       warning: method formatted in class StringFormat is deprecated (since 2.12.16): Use `formatString.format(value)` instead of `value.formatted(formatString)`,
       or use the `f""` string interpolator. In Java 15 and later, `formatted` resolves to the new method in String which has reversed parameters.
val res0: String = 42

scala> 42.formatted()
                   ^
       error: not enough arguments for method formatted: (fmtstr: String): String.
       Unspecified value parameter fmtstr.
```

## Problem

Although the general rule is to show errors first and then warnings, on the assumption that errors cascade in spurious warnings, even though warnings may reveal what leads to the error, the descriptive text for a symbol in an error message should include whether it is deprecated.

Possibly it is required only when `-deprecation` is enabled.

### PR DESCRIPTION
**Pull request by som-snytt**
Fixes scala/bug#12730

### CODE REVIEW

**Comment by lrytz** [comment]
@SethTisue wdyt?

**Comment by SethTisue** [comment]
Obviously the goal here is to be helpful to the user... Will it be helpful some of the time? I suppose it will. But I'm not sure it's worth the costs. I see two costs:

- General verbosity, which is annoying in situations where you don't want the extra nag
- Inconsistency as to which error messages make reference to a symbol's deprecated status and which don't

The number of affected checkfiles is pretty small, which suggests to me that we're creating a situation here where from the user's perspective, the compiler will occasionally mention that something is deprecated, with no apparent rhyme or reason to why it's being mentioned or not.

Basically, I think it's the warnings system’s job to alert users when deprecated API is involved, and we don't need other mechanisms trying to pitch in and help.

As to the argument made on scala/bug#12730... well, it seems normal to me that working with the compiler goes in two phases, one where you get the damn thing to compile, and a second where you deal with warnings if there are any.

Open to discussion of course, but that's my initial reaction.

**Comment by som-snytt** [comment]
This was an experiment, but I like the outcome.

If I read a diagnostic about `ident`, then its deprecation status is relevant. It's like murder during the commission of a crime. Although the burglary is the primary offense, it's relevant if someone died, because it's a felony. You would never mention the burglary without adding that someone was murdered.

I think `deprecated` is as cogent as verbiage such as `value` (recent wordle, by the way; I got it in two).

```
warning: value f is deprecated
```
There is no possible ambiguity because the caret is pointing at `f`; the word "value" could be omitted without loss of clarity.

Alternative renderings of `deprecated`, a long word, would be strike-through or BOLD RED. Or an addendum:
```
class C is deprecated. Are you sure you know what you're doing?
```

In a similar vein, dotty still says
```
1 warning found
```
where "found" has been dropped in scala 2. It's just a filler word.

**Comment by som-snytt** [comment]
I like the addendum, which I may try out.

**Comment by som-snytt** [comment]
I will update with highlighting instead. Everyone makes fun of scala 2 for not highlighting.

**Comment by SethTisue** [comment]
The highlighting idea I like even less. I really don't think this is desirable in any form.

**Comment by som-snytt** [comment]
This can wait for `-explain` awesome messages. I see in May I suggested an addendum.

Closing again due to vehemence.
